Library/Homebrew/formula.rb

```diff
@@ -1494,10 +1495,20 @@ def runtime_dependencies(read_from_tab: true)
       return tab_deps.map { |d| Dependency.new d["full_name"] }.compact
     end
 
-    recursive_dependencies do |_, dependency|
-      Dependency.prune if dependency.build?
-      Dependency.prune if !dependency.required? && build.without?(dependency)
+    declared_runtime_dependencies | undeclared_runtime_dependencies
+  end
+
+  def undeclared_runtime_dependencies
+    if optlinked?
+      keg = Keg.new(opt_prefix)
```

Comment:
What's the advantage of using the `opt_prefix` when it's `optlinked?`?

Reply:
Uses the currently active version of a formula rather than the latest. Can happen with `brew switch` or even just if `brew upgrade` hasn't been run yet.